### PR TITLE
pythonPackages: Add OpenStack Octavia client and Neutron bindings

### LINKED ISSUE #320529: Package request: python-octaviaclient
**Project description**

Octavia client for OpenStack Load Balancing

This is an OpenStack Client (OSC) plugin for Octavia, an OpenStack Load Balancing project.

This depends upon nixpkg: `python-openstackclient`.

**Metadata**

* homepage URL: https://docs.openstack.org/python-octaviaclient/latest/
* source URL: https://opendev.org/openstack/python-octaviaclient
* license: apache
* platforms: python / cross platform

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by Vinetos**
## Description of changes

Add python-octaviaclient in nix ecosystem.   
The documentation cannot be built for now as `sphinxcontrib-svg2pdfconverter` does not exist yet.  
I am marking this MR as Draft as I am not fully confident about this.

Closes #320529

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Vinetos** [comment]
I am not able to run commands as `openstack loadbalancer ...`. Is there anything wrong here ?

**Comment by Vinetos** [comment]
I saw that OpenCtack client use the entrypoint mecanism to discover and load extension.

File: pkgs/development/python-modules/python-octaviaclient/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/python-octaviaclient/default.nix:16]:
```suggestion
```

File: pkgs/development/python-modules/python-octaviaclient/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/python-octaviaclient/default.nix:22]:
```suggestion
  pyproject = true;
```

File: pkgs/development/python-modules/python-octaviaclient/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/python-octaviaclient/default.nix:63]:
```suggestion
```

**Comment by SuperSandro2000** [comment]
> I saw that OpenCtack client use the entrypoint mecanism to discover and load extension.

I haven't worry about that yet.

> I have a lot to learn about Python packaging so, I do not know if I can but as your the the last member of the OpenStack team,

We could probably add a top-level entry similar to swift.

**Comment by Vinetos** [comment]
> We could probably add a top-level entry similar to swift.  

For the package or for the team ? or both ?

**Comment by Vinetos** [comment]
Another thing : I have to package python-neutronclient and os-client-config. Both has been deprecated and only used for the bindings, should I mark them as deprecated in nixpkgs too ?

**Comment by RossComputerGuy** [comment]
What's with the Flutter commits?

**Comment by Vinetos** [comment]
Oops; I have just messed up with my git history, mb

**Comment by Vinetos** [comment]
Waits for #326279 as octaviaclient depends on doc8

File: pkgs/development/python-modules/os-client-config/default.nix
Comment (anthonyroussel) [inline pkgs/development/python-modules/os-client-config/default.nix:17]:
```suggestion

```

File: pkgs/development/python-modules/os-client-config/default.nix
Comment (anthonyroussel) [inline pkgs/development/python-modules/os-client-config/default.nix:27]:
```suggestion
  build-system = [ setuptools ];

  dependencies = [
    glanceclient
  ];
```

We can use build-system and dependencies here.

File: pkgs/development/python-modules/python-neutronclient/default.nix
Comment (anthonyroussel) [inline pkgs/development/python-modules/python-neutronclient/default.nix:33]:
```suggestion
  build-system = [
    setuptools
  ];

  dependencies = [
    osc-lib
    openstackclient
    os-client-config
  ];

```

**Comment by anthonyroussel** [comment]
@ofborg build python312Packages.os-client-config python312Packages.python-neutronclient python312Packages.python-octaviaclient python311Packages.os-client-config python311Packages.python-neutronclient python311Packages.python-octaviaclient

**Comment by anthonyroussel** [comment]
To trigger the ofborg CI, you need to rename your commits like these (just an example):

`pythonPackages.os-client-config: init at 2.1.0` > `python312Packages.os-client-config: init at 2.1.0`

See https://github.com/NixOS/ofborg?tab=readme-ov-file#automatic-building

File: pkgs/development/python-modules/python-neutronclient/default.nix
Comment (anthonyroussel) [inline pkgs/development/python-modules/python-neutronclient/default.nix:37]:
```suggestion
    oslo-log
    stestr
```

stestr is missing

Comment (anthonyroussel) [reply pkgs/development/python-modules/python-neutronclient/default.nix:37]: osprofiler is also a required python dependency needed for the tests
But looks like there is no package for it in NixOS: https://opendev.org/openstack/python-neutronclient/src/branch/master/test-requirements.txt#L8

File: pkgs/development/python-modules/os-client-config/default.nix
Comment (Vinetos) [reply pkgs/development/python-modules/os-client-config/default.nix:17]: Thanks your big review !  
I am sorry you have to go through all of these files and found a lot because I forgot to push the latest commits I have done for this MR !

File: pkgs/development/python-modules/os-client-config/default.nix
Comment (Vinetos) [reply pkgs/development/python-modules/os-client-config/default.nix]: Should I use dependencies for this package or could I keep the one I have pushed ?  
What is the difference and how should I do ?

**Comment by Vinetos** [comment]
Hey @anthonyroussel 
Could you review this MR again ? 
I forgot to push my latest changes but some of your points are still valid.  
When should I use `dependencies` instead of `propagatedBuildInputs` ? Should, in python packing, `propagatedBuildInputs` be used ?
Thanks !
Val

**Comment by Vinetos** [comment]
<details> 
  <summary>14 packages built:</summary>
python311Packages.doc8 python311Packages.doc8.dist python311Packages.os-client-config python311Packages.os-client-config.dist python311Packages.osprofiler python311Packages.osprofiler.dist python311Packages.python-neutronclient python311Packages.python-neutronclient.dist python311Packages.python-octaviaclient python311Packages.python-octaviaclient.dist python311Packages.tempest python311Packages.tempest.dist python312Packages.doc8 python312Packages.doc8.dist
</details>

File: pkgs/development/python-modules/os-client-config/default.nix
Comment (anthonyroussel) [reply pkgs/development/python-modules/os-client-config/default.nix]: `glanceclient` is a Python runtime dependency, so it should go in `dependencies`

`build-system` and `dependencies` were introduced recently to separate runtime & build time Python dependencies, and solve dependency propagation in Python.

> When should I use dependencies instead of propagatedBuildInputs ? Should, in python packing, propagatedBuildInputs be used ?

If I am not mistaken,

* `build-system` is for build-time Python dependencies only (`setup_requires`)
* `dependencies` is for runtime Python dependencies only (`install_requires`)
* `propagatedBuildInputs` can be used if your package relies on a runtime dependency that is not a Python dependency for a Python package

You can find more documentation/information here:
* https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters
* https://github.com/NixOS/nixpkgs/pull/271597

**Comment by anthonyroussel** [comment]
Result of `nixpkgs-review pr 322593` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>python311Packages.doc8</li>
    <li>python311Packages.doc8.dist</li>
    <li>python311Packages.os-client-config</li>
    <li>python311Packages.os-client-config.dist</li>
    <li>python311Packages.osprofiler</li>
    <li>python311Packages.osprofiler.dist</li>
    <li>python311Packages.python-neutronclient</li>
    <li>python311Packages.python-neutronclient.dist</li>
    <li>python311Packages.python-octaviaclient</li>
    <li>python311Packages.python-octaviaclient.dist</li>
    <li>python311Packages.tempest</li>
    <li>python311Packages.tempest.dist</li>
    <li>python312Packages.doc8</li>
    <li>python312Packages.doc8.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/python-octaviaclient/default.nix
Comment (anthonyroussel) [inline pkgs/development/python-modules/python-octaviaclient/default.nix:6]:
```suggestion
```

Unused

File: pkgs/development/python-modules/python-octaviaclient/default.nix
Comment (anthonyroussel) [inline pkgs/development/python-modules/python-octaviaclient/default.nix:47]:
```suggestion
```

`setuptools` is already in `build-system`

File: pkgs/development/python-modules/python-neutronclient/default.nix
Comment (anthonyroussel) [inline pkgs/development/python-modules/python-neutronclient/default.nix:1]: By the way, do we really need the python-neutronclient package in NixOS (and the dependencies on os-client-config and osprofiler packages)?

Because since the OpenStack Zed release, python-neutronclient has been deprecated and the code has been migrated to [OpenStack Client](https://opendev.org/openstack/python-openstackclient) (OCS).

See

* https://lists.openstack.org/archives/list/openstack-discuss@lists.openstack.org/message/KTTBW2DNGELWHKOXMMBAQYTKKQDX2VNF/
* https://docs.openstack.org/python-neutronclient/latest/contributor/transition_to_osc.html

Comment (Vinetos) [reply pkgs/development/python-modules/python-neutronclient/default.nix:1]: I have followed the [requirements.txt](https://opendev.org/openstack/python-octaviaclient/src/branch/master/requirements.txt#L11) of octaviaclient. 

I do not have checked octaviaclient but it may use some old bindings performed by the neutron client

I will check this day

File: pkgs/development/python-modules/os-client-config/default.nix
Comment (Vinetos) [reply pkgs/development/python-modules/os-client-config/default.nix]: Thanks for the explanation. I'm gonna update the MR then

File: pkgs/development/python-modules/python-neutronclient/default.nix
Comment (Vinetos) [reply pkgs/development/python-modules/python-neutronclient/default.nix:1]: neutronclient is still used for the bindings in octaviaclient.  
There are calls in order to list networks, subnets, ports and qos policies on load balancers
https://opendev.org/openstack/python-octaviaclient/src/branch/master/octaviaclient/osc/v2/utils.py#L192

**Comment by Vinetos** [comment]
14 packages built :smile:

File: pkgs/development/python-modules/doc8/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/doc8/default.nix:12]:
```suggestion
```
We no longer need to add `pythonRelaxDepsHook` explicitly.

File: pkgs/development/python-modules/doc8/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/doc8/default.nix:43]:
```suggestion
```

File: pkgs/development/python-modules/os-client-config/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/os-client-config/default.nix:39]:
```suggestion
```
We don't need to measure the coverage.

File: pkgs/development/python-modules/os-client-config/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/os-client-config/default.nix:51]:
```suggestion
  checkPhase = ''
    runHook preCheck

    stestr run

    runHook postCheck
  '';
```

File: pkgs/development/python-modules/python-neutronclient/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/python-neutronclient/default.nix:52]: https://opendev.org/openstack/python-neutronclient/src/commit/a10390c62e3405ae604206b60bca4703fda8da2c/setup.py#L19
`pbr` should go to `build-system`.

File: pkgs/development/python-modules/python-neutronclient/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/python-neutronclient/default.nix:74]:
```suggestion
```

File: pkgs/development/python-modules/python-neutronclient/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/python-neutronclient/default.nix:88]:
```suggestion
  checkPhase = ''
    runHook preCheck

    stestr run

    runHook postCheck
  '';
```

File: pkgs/development/python-modules/python-neutronclient/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/python-neutronclient/default.nix:72]:
```suggestion
```

File: pkgs/development/python-modules/python-octaviaclient/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/python-octaviaclient/default.nix:59]: `pbr` should go to `build-system`.

File: pkgs/development/python-modules/python-octaviaclient/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/python-octaviaclient/default.nix:66]:
```suggestion
```

File: pkgs/development/python-modules/python-octaviaclient/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/python-octaviaclient/default.nix:72]:
```suggestion
```

File: pkgs/development/python-modules/python-octaviaclient/default.nix
Comment (natsukium) [inline pkgs/development/python-modules/python-octaviaclient/default.nix:81]:
```suggestion
  checkPhase = ''
    runHook preCheck

    stestr run

    runHook postCheck
  '';
```

**Comment by Vinetos** [comment]
Result of `nixpkgs-review pr 322593` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>python311Packages.doc8</li>
    <li>python311Packages.doc8.dist</li>
    <li>python311Packages.os-client-config</li>
    <li>python311Packages.os-client-config.dist</li>
    <li>python311Packages.osprofiler</li>
    <li>python311Packages.osprofiler.dist</li>
    <li>python311Packages.python-neutronclient</li>
    <li>python311Packages.python-neutronclient.dist</li>
    <li>python311Packages.python-octaviaclient</li>
    <li>python311Packages.python-octaviaclient.dist</li>
    <li>python311Packages.tempest</li>
    <li>python311Packages.tempest.dist</li>
    <li>python312Packages.doc8</li>
    <li>python312Packages.doc8.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/python-neutronclient/default.nix
Comment (anthonyroussel) [reply pkgs/development/python-modules/python-neutronclient/default.nix]: Oops sorry, indeed octaviaclient depends on neutronclient package.

**Comment by anthonyroussel** [comment]
Result of `nixpkgs-review pr 322593` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>python311Packages.doc8</li>
    <li>python311Packages.doc8.dist</li>
    <li>python311Packages.os-client-config</li>
    <li>python311Packages.os-client-config.dist</li>
    <li>python311Packages.osprofiler</li>
    <li>python311Packages.osprofiler.dist</li>
    <li>python311Packages.python-neutronclient</li>
    <li>python311Packages.python-neutronclient.dist</li>
    <li>python311Packages.python-octaviaclient</li>
    <li>python311Packages.python-octaviaclient.dist</li>
    <li>python311Packages.tempest</li>
    <li>python311Packages.tempest.dist</li>
    <li>python312Packages.doc8</li>
    <li>python312Packages.doc8.dist</li>
  </ul>
</details>

**Comment by Vinetos** [comment]
@SuperSandro2000 as the last member of the team, do you allow me to commit my addition to the OpenStack team?  
As I use OS daily, I want to add more packages and help maintaining the other.

**Review by anthonyroussel — APPROVED** [review]

**Comment by anthonyroussel** [comment]
Hello @Vinetos 

Could you please add `python-neutronclient` and `python-octaviaclient` to `passthru.optional-dependencies.cli-plugins` in the pkgs/development/python-modules/python-openstackclient/default.nix file, please? :)

If you add your 2 new OpenStack CLI plugin packages inside this list, it will build the `openstackclient-full` package which is the OpenStack Client with all the CLI plugins enabled.

```diff
diff --git a/pkgs/development/python-modules/python-openstackclient/default.nix b/pkgs/development/python-modules/python-openstackclient/default.nix
index e3dd953e93b0..90ebcbe8ed6a 100644
--- a/pkgs/development/python-modules/python-openstackclient/default.nix
+++ b/pkgs/development/python-modules/python-openstackclient/default.nix
@@ -13,7 +13,9 @@
   python-ironicclient,
   python-keystoneclient,
   python-manilaclient,
+  python-neutronclient,
   python-novaclient,
+  python-octaviaclient,
   python-openstackclient,
   requests-mock,
   setuptools,
@@ -73,6 +75,8 @@ buildPythonPackage rec {
         python-heatclient
         python-ironicclient
         python-manilaclient
+        python-neutronclient
+        python-octaviaclient
       ];
     };
     tests.version = testers.testVersion {
```

**Comment by Vinetos** [comment]
> Could you please add `python-neutronclient` and `python-octaviaclient` to `passthru.optional-dependencies.cli-plugins` in the pkgs/development/python-modules/python-openstackclient/default.nix file, please? :)
> 

Done !

> If you add your 2 new OpenStack CLI plugin packages inside this list, it will build the `openstackclient-full` package with your new packages which is the OpenStack Client with all the CLI plugins enabled.

That's a very cool addition ! :smile:

File: pkgs/development/python-modules/python-openstackclient/default.nix
Comment (anthonyroussel) [inline pkgs/development/python-modules/python-openstackclient/default.nix:79]:
```suggestion
```

Sorry I said a mistake, it cannot work with python-octaviaclient currently.
Because we have an infinite recursion here when building openstackclient-full which depends on python-octaviaclient which depends on openstackclient, which depends on python-octaviaclient...

File: pkgs/development/python-modules/python-neutronclient/default.nix
Comment (anthonyroussel) [inline pkgs/development/python-modules/python-neutronclient/default.nix:73]:
```suggestion
    python-openstackclient
```

Should be the library python-openstackclient

**Comment by Vinetos** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>openstackclient-full</li>
    <li>openstackclient-full.dist</li>
    <li>python311Packages.doc8</li>
    <li>python311Packages.doc8.dist</li>
    <li>python311Packages.os-client-config</li>
    <li>python311Packages.os-client-config.dist</li>
    <li>python311Packages.osprofiler</li>
    <li>python311Packages.osprofiler.dist</li>
    <li>python311Packages.python-neutronclient</li>
    <li>python311Packages.python-neutronclient.dist</li>
    <li>python311Packages.python-octaviaclient</li>
    <li>python311Packages.python-octaviaclient.dist</li>
    <li>python312Packages.doc8</li>
    <li>python312Packages.doc8.dist</li>
  </ul>
</details>

**Review by anthonyroussel — APPROVED** [review]

**Comment by anthonyroussel** [comment]
@Vinetos Hello!

Unfortunately, there are some merge conflicts with your branch. 

Once these merge conflicts are resolved, maybe we can try to submit your PR in [the PRs ready for review Discourse thread](https://discourse.nixos.org/t/prs-ready-for-review/3032) to try to find someone with merge rights.

File: pkgs/development/python-modules/osprofiler/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/osprofiler/default.nix:27]: We should add oslo-utils, too. In the next version we also need some others https://opendev.org/openstack/osprofiler/src/tag/4.1.0/requirements.txt

**Review by SuperSandro2000 — COMMENTED** [review]
otherwise LGTM and I can merge it

**Comment by Vinetos** [comment]
Ready :D

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>openstackclient-full</li>
    <li>openstackclient-full.dist</li>
    <li>python311Packages.doc8</li>
    <li>python311Packages.doc8.dist</li>
    <li>python311Packages.os-client-config</li>
    <li>python311Packages.os-client-config.dist</li>
    <li>python311Packages.osprofiler</li>
    <li>python311Packages.osprofiler.dist</li>
    <li>python311Packages.python-neutronclient</li>
    <li>python311Packages.python-neutronclient.dist</li>
    <li>python311Packages.python-octaviaclient</li>
    <li>python311Packages.python-octaviaclient.dist</li>
    <li>python312Packages.doc8</li>
    <li>python312Packages.doc8.dist</li>
  </ul>
</details>

**Review by anthonyroussel — APPROVED** [review]
Result of `nixpkgs-review pr 322593` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>openstackclient-full</li>
    <li>openstackclient-full.dist</li>
    <li>python311Packages.doc8</li>
    <li>python311Packages.doc8.dist</li>
    <li>python311Packages.os-client-config</li>
    <li>python311Packages.os-client-config.dist</li>
    <li>python311Packages.osprofiler</li>
    <li>python311Packages.osprofiler.dist</li>
    <li>python311Packages.python-neutronclient</li>
    <li>python311Packages.python-neutronclient.dist</li>
    <li>python311Packages.python-octaviaclient</li>
    <li>python311Packages.python-octaviaclient.dist</li>
    <li>python312Packages.doc8</li>
    <li>python312Packages.doc8.dist</li>
  </ul>
</details>

---

Result of `nixpkgs-review pr 322593` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>openstackclient-full</li>
    <li>openstackclient-full.dist</li>
    <li>python311Packages.doc8</li>
    <li>python311Packages.doc8.dist</li>
    <li>python311Packages.os-client-config</li>
    <li>python311Packages.os-client-config.dist</li>
    <li>python311Packages.osprofiler</li>
    <li>python311Packages.osprofiler.dist</li>
    <li>python311Packages.python-neutronclient</li>
    <li>python311Packages.python-neutronclient.dist</li>
    <li>python311Packages.python-octaviaclient</li>
    <li>python311Packages.python-octaviaclient.dist</li>
    <li>python312Packages.doc8</li>
    <li>python312Packages.doc8.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/python-openstackclient/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/python-openstackclient/default.nix:17]: novaclient is not used

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**
